File: Ural/2023. Donald is a postman.java
import java.io.PrintWriter;
import java.util.HashMap;
import java.util.Scanner;

public class ACM
{
	public static void main(String[] args)
	{
		Scanner in = new Scanner(System.in);
		PrintWriter out = new PrintWriter(System.out);
		
		HashMap<Character, Integer> map = new HashMap<Character, Integer>();
		map.put('A', 1);
		map.put('P', 1);
		map.put('O', 1);
		map.put('R', 1);
		map.put('B', 2);
		map.put('M', 2);
		map.put('S', 2);
		map.put('D', 3);
		map.put('G', 3);
		map.put('J', 3);
		map.put('K', 3);
		map.put('T', 3);
		map.put('W', 3);

		int currentPos = 1; 
		int ret = 0; 
		int n = in.nextInt();
		for(int i = 1; i <= n; i++){
			String str = in.next();
			int pos = map.get(str.charAt(0));
			ret += Math.abs(currentPos - pos);
			currentPos = pos; 
		}
		out.println(ret);
		out.flush();
		in.close();
		out.close(); 
	}
};

File: Ural/2023. Donald is a postman.cpp


#include <iostream>
#include <unordered_map>
#include <string>
#include <cmath>

using namespace std; 

int main(){
	int n = 0;
	int ret = 0; 
	unordered_map<char, int> map; 
	map['A'] = 1;
	map['P'] = 1;
	map['O'] = 1;
	map['R'] = 1;
	map['B'] = 2;
	map['M'] = 2;
	map['S'] = 2;
	map['D'] = 3;
	map['G'] = 3;
	map['J'] = 3;
	map['K'] = 3;
	map['T'] = 3;
	map['W'] = 3;
	cin >> n; 
	string str; 
	int currentPos = 1; 
	for(auto i = 1; i <= n; i++){
		cin >> str; 
		int pos = map[str[0]];
		ret += abs(currentPos - pos);
		currentPos = pos; 
	}
	cout << ret << endl;
	return 0; 
}
